xline/src/storage/compact/periodic_compactor.rs

```diff
@@ -186,12 +177,20 @@ impl<C: Compactable> Compactor for PeriodicCompactor<C> {
                 }
                 // To ensure that each iteration invokes the same `shutdown_trigger` and keeps
                 // events losing due to the cancellation of `shutdown_trigger` at bay.
-                _ = &mut shutdown_trigger => {
+                _ = shutdown_listener.wait_self_shutdown() => {
```

Comment:
Is it possible for this to lose notifications? And please remove or update the comments above.

Reply:
Removed